File: box_office_app_node/queries.js


const db_info= {
    // user: 'postgres',
    user: 'edward',
    // host: 'db',
    host: 'localhost',
    database: 'postgres',
    password: 'mysecretpassword',
    port: '5432'
}

async function query(text, params,pool){
    return await pool.query(text, params);
}




module.exports = {query,db_info};

File: box_office_app_flask/queries.py
from psycopg_pool import AsyncConnectionPool
import asyncio

connection_str = ''.join(['user=edward host=localhost dbname=postgres ',
'password=mysecretpassword port=5432'])


async def create_pool():
    pool = AsyncConnectionPool(connection_str,open=False)
    await pool.open(wait=True)
    return  pool
    
async def query(text,params,pool):
    async with pool.connection() as conn:
        async with conn.cursor() as cur:
            await cur.execute(text, params)
            return await cur.fetchall()



